warc_metadata_sidecar.py

```diff
@@ -207,18 +225,26 @@ def metadata_sidecar(archive_dir, warc_file, operator=None, publisher=None):
             print(url)
             payload.seek(0)
             mime_dict, puid = find_mime_and_puid(fido, payload)
+            mimes_found = ' '.join(mime_dict.values())
+            soft404_detected = None
             result_dict = {}
             lang_cld = None
+            # Determine the soft404 probability on html records.
+            status = record.http_headers.get_statuscode()
+            if status == '200' and 'html' in mimes_found:
+                payload.seek(0)
+                soft404_detected = determine_soft404(payload.read())
```

Comment:
It seems like we could move the soft-404 code inside of the `if TEXT_FORMAT_MIMES.search(mimes_found)`  block, since 'html' should only happen as a sub-case of this, and we can pass `bytes_payload` to `determine_soft404` then.